### PR TITLE
[ADD] web_dashboard: added web_dashboard module (for browsing modules, invite new user quickly, access progress of all planners)

### PR DESCRIPTION
**Pull request by pga-odoo**
[ADD] added web_dashboard module
[ADD] added video and share dashboard
[IMP] solved planner dashboard issues occurred due to change in planner in master
[ADD] added dashboard data file
[IMP]added asset priority because it depends on planner
[IMP] improved font-size, resend button removed and minor changes
[FIX] solved small screen issue with media css
[IMP] removed extra custom style and added videos
[IMP] improved btn margins, added text-muted and some minor changes
[FIX] fixed wrong less, text-muted
[IMP] added links border for better visuals
[IMP] improved code according new JavaScript Format
[IMP] improved web_dashboard
[IMP] it's all about the space
[IMP] removed custom loader, bypass browser's popup blocker
[IMP] Improved rendered using deferred, reloaded on do_show
[IMP] improved code according to new ActionManager Fixes, and removed do_show
[IMP] moved email validation in JS, improved less
[IMP] improved summary
[IMP] Removed video dashboard and added amazone gift dashboard
[IMP] Added config redirect warning if outgoing mail server not configured
[IMP] Improved Share Dashbaord, used zeroclipboard for copy url
[IMP] added Gift card url
[ADD] added social share dashboard
[FIX] fixed user delete historyback issue

### CODE REVIEW

File: addons/web_dashboard/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            },
+            'gift': {
+                'user_email': request.env.user.email
+            }
+        }
+
+    @http.route('/dashboard/create_users', type='json', auth='user')
+    def create_user(self, email_adderesses, optional_message=""):
+        Users = request.env['res.users']
+        email_adderesses = map(str.strip, email_adderesses)
```
Comment (rga-odoo) [inline addons/web_dashboard/controllers/main.py]: traceback: 
    email_adderesses = map(str.strip, email_adderesses)
TypeError: descriptor 'strip' requires a 'str' object but received a 'unicode'

Comment (pga-odoo) [inline addons/web_dashboard/controllers/main.py]: I'm not getting traceback on user creation from dashboard. Can you tell me steps to reproduce ??

Comment (rga-odoo) [inline addons/web_dashboard/controllers/main.py]: oops my bad I tested it against python-dbg, it quite strange that simplejson.loads return unicode string while without debug mode it return string

Comment (pga-odoo) [inline addons/web_dashboard/controllers/main.py]: :joy: :+1:

File: addons/web_dashboard/static/src/js/dashboard.js [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+        var user_emails = _.str.trim(this.$('#user_emails').val(), '\n').split("\n");
+        var optional_message = this.$('#optional_message').val();
+        var re = /^([\w-]+(?:\.[\w-]+)*)@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$/i;
+        var is_valid_emails = _.every(user_emails, function(email) {
+            return re.test(email);
+        });
+        if(is_valid_emails){
+            $target.prop('disabled', true);
+            $target.find('i.fa-cog').removeClass('hidden');
+            session.rpc("/dashboard/create_users", {
```
Comment (antonylesuisse) [inline addons/web_dashboard/static/src/js/dashboard.js]: Why a conttrooler when you can juste Model('res.users').call('serach')

Just call a method on the right objects.

File: addons/web_dashboard/controllers/main.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+class Dashboard(http.Controller):
+
+    def get_users_list(self):
+        users = request.env['res.users'].search([('login_date', '=', False)], order="create_date desc")
+        pending = users.filtered(lambda u: u.signup_valid)
+        expired = users - pending
+        return {'expired': zip(expired.mapped('id'), expired.mapped('login')), 'pending': zip(pending.mapped('id'), pending.mapped('login'))}
+
+    @http.route('/dashboard/info', type='json', auth='user')
+    def get_info(self, **kw):
```
Comment (antonylesuisse) [inline addons/web_dashboard/controllers/main.py]: maybe ok

File: addons/web_dashboard/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+from openerp import http, _
+from openerp.http import request
+from openerp.exceptions import RedirectWarning
+from openerp.tools.misc import DEFAULT_SERVER_DATETIME_FORMAT
+
+
+class Dashboard(http.Controller):
+
+    def get_users_list(self):
```
Comment (antonylesuisse) [inline addons/web_dashboard/controllers/main.py]: call res.users (if you need you add a mtehod on res.users ) web_dashboard_get_XXXXX

File: addons/web_dashboard/controllers/main.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                'planners': planners,
+                'overall_progress': sum([p['progress'] for p in planners]) / (len(planners) or 1)
+            },
+            'gift': {
+                'user_email': request.env.user.email
+            }
+        }
+
+    @http.route('/dashboard/create_users', type='json', auth='user')
+    def create_user(self, email_adderesses, optional_message=""):
```
Comment (antonylesuisse) [inline addons/web_dashboard/controllers/main.py]: move it to res.users

File: addons/auth_signup/auth_signup_data.xml [hunk 6]
```diff
@@ -50,6 +50,11 @@
                     <p>
                         You have been invited to connect to "${object.company_id.name}" in order to get access to your documents in Odoo.
                     </p>
+                    % if ctx.get("custom_message"):
+                    <p>
+                        ${ctx.custom_message}
```
Comment (tde-banana-odoo) [inline addons/auth_signup/auth_signup_data.xml]: ctx['custom_message'] ?

File: addons/auth_signup/res_users.py [hunk 7]
```diff
@@ -194,7 +195,8 @@ def signup(self, cr, uid, values, token=None, context=None):
             if partner_user:
                 # user exists, modify it according to values
                 values.pop('login', None)
-                values.pop('name', None)
+                if single_email_re.match(partner_user.name) == None:
```
Comment (tde-banana-odoo) [inline addons/auth_signup/res_users.py]: 1/ Why ? 2/ Once you explained why, put in a separate commit with the why :) .

File: addons/auth_signup/views/auth_signup_login.xml [hunk 8]
```diff
@@ -20,7 +20,7 @@
             <div class="form-group field-name">
                 <label for="name" class="control-label">Your Name</label>
                 <input type="text" name="name" t-att-value="name" id="name" class="form-control" placeholder="e.g. John Doe"
-                    required="required" t-att-readonly="'readonly' if only_passwords else None"
+                    required="required" t-att-readonly="'readonly' if (only_passwords and name != login) else None"
```
Comment (tde-banana-odoo) [inline addons/auth_signup/views/auth_signup_login.xml]: 1/ Why ? 2/ Once you explained why, put in a separate commit with the why :) .

File: addons/web_dashboard/models/res_users.py [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+    def web_dashboard_get_users_by_state(self):
+        users = self.search([('login_date', '=', False)], order="create_date desc")
+        pending = users.filtered(lambda u: u.signup_valid)
+        expired = users - pending
+        return {'expired': zip(expired.mapped('id'), expired.mapped('login')), 'pending': zip(pending.mapped('id'), pending.mapped('login'))}
+
+    @api.model
+    def web_dashboard_create_users(self, email_adderesses, optional_message=""):
+        email_adderesses = map(str.strip, email_adderesses)
```
Comment (tde-banana-odoo) [inline addons/web_dashboard/models/res_users.py]: In tools there are already existing stuff to extract email addresses from text.

File: addons/web_dashboard/models/res_users.py [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    _inherit = 'res.users'
+
+    def web_dashboard_get_users_by_state(self):
+        users = self.search([('login_date', '=', False)], order="create_date desc")
+        pending = users.filtered(lambda u: u.signup_valid)
+        expired = users - pending
+        return {'expired': zip(expired.mapped('id'), expired.mapped('login')), 'pending': zip(pending.mapped('id'), pending.mapped('login'))}
+
+    @api.model
+    def web_dashboard_create_users(self, email_adderesses, optional_message=""):
```
Comment (tde-banana-odoo) [inline addons/web_dashboard/models/res_users.py]: Please read what you type: adderesse does not exists in English. Maybe you want to say adresse like in French, but I don't think so, actually :) .

File: addons/web_dashboard/models/res_users.py [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+from openerp import models, api, _
+from openerp.exceptions import RedirectWarning
+from openerp.tools.misc import DEFAULT_SERVER_DATETIME_FORMAT
+
+
+class Users(models.Model):
+    _inherit = 'res.users'
+
+    def web_dashboard_get_users_by_state(self):
```
Comment (tde-banana-odoo) [inline addons/web_dashboard/models/res_users.py]: I think @antonylesuisse asked you to make a method on res.users... but for me this one belongs more to a controller (only marshmalling).

File: addons/web_dashboard/models/res_users.py [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+        users = self.search([('login_date', '=', False)], order="create_date desc")
+        pending = users.filtered(lambda u: u.signup_valid)
+        expired = users - pending
+        return {'expired': zip(expired.mapped('id'), expired.mapped('login')), 'pending': zip(pending.mapped('id'), pending.mapped('login'))}
+
+    @api.model
+    def web_dashboard_create_users(self, email_adderesses, optional_message=""):
+        email_adderesses = map(str.strip, email_adderesses)
+        existing_users = self.search([('login', 'in', email_adderesses), '|', ('active', '=', True), ('active', '=', False)])
```
Comment (tde-banana-odoo) [inline addons/web_dashboard/models/res_users.py]: Implementation detail: add an '&' at the beginning of the domain (explicit better than implicit).

File: addons/web_dashboard/models/res_users.py [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+        #Processing unused email adresses : create new users (Used template user)
+        template_user_id = self.env['ir.config_parameter'].get_param('auth_signup.template_user_id', 'False')
+        for email in unused_emails:
+            default_values = {'login': email, 'name': email, 'email': email, 'active': True}
+            if template_user_id:
+                user = self.with_context(custom_message=optional_message, create_user=1).browse(int(template_user_id)).copy(default=default_values)
+            else:
+                user = self.with_context(custom_message=optional_message, create_user=1).create(default_values)
+            # if user haven't confugered Outgoing mail server
+            if user.signup_token == False:
```
Comment (tde-banana-odoo) [inline addons/web_dashboard/models/res_users.py]: signup_token is False -> automatically linked to the outgoing mail server ? Are you sure ?

File: addons/web_dashboard/controllers/main.py [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+from openerp import http
+from openerp.http import request
+
+
+class Dashboard(http.Controller):
+
+    @http.route('/dashboard/info', type='json', auth='user')
+    def get_info(self, **kw):
+        installed_apps = request.env['ir.module.module'].search_count([('application', '=',  True), ('state', 'in', ['installed', 'to upgrade', 'to remove'])])
+        active_users = request.env['res.users'].search_count([('active', '=', True), ('login_date', '!=', False)])
```
Comment (tde-banana-odoo) [inline addons/web_dashboard/controllers/main.py]: Just asking: what about portal users (share flag) ?

File: addons/auth_signup/res_users.py (hunk 7 above)
Comment (pga-odoo) [inline addons/auth_signup/res_users.py]: From dashboard we are inviting user just by email and name and login are required so i have to put email in both fields. So when user sign up he can change his name (only if name is email id).  
download 7


download 5

File: addons/auth_signup/auth_signup_data.xml (hunk 6 above)
Comment (pga-odoo) [inline addons/auth_signup/auth_signup_data.xml]: We are not saving message in database. so i sent it through context.
download 8

File: addons/web_dashboard/controllers/main.py (hunk 14 above)
Comment (pga-odoo) [inline addons/web_dashboard/controllers/main.py]: Invited user by default having access rights of `Template User` (auth_signup.template_user_id). So i included portal user as a active users.

File: addons/web_dashboard/models/res_users.py (hunk 13 above)
Comment (pga-odoo) [inline addons/web_dashboard/models/res_users.py]: After creating user if mail server not configured (after mail Exception)  signup_token is set to false.

File: addons/auth_signup/res_users.py (hunk 7 above)
Comment (pga-odoo) [inline addons/auth_signup/res_users.py]: @tde-banana-odoo  should i put it in separate commit ??

File: addons/web_dashboard/static/src/js/dashboard.js [hunk 15]
```diff
@@ -0,0 +1,264 @@
...
+            });
+            $.when.apply($, deferred_promises)
+                .then(function() {
+                    loading_done.resolve();
+                });
+        });
+        return loading_done;
+    },
+    apps: function(data){
+        this.$('.o_web_dashboard_apps').append(QWeb.render("DashboardApps", data['apps']));
```
Comment (mge-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: @pga-odoo  Why is this one different from the 4 others ?

Comment (mge-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: @pga-odoo  Why is this one different from the 4 others ?

File: addons/web_dashboard/static/src/js/dashboard.js [hunk 16]
```diff
@@ -0,0 +1,264 @@
...
+                    self.reload();
+                });
+
+        }else{
+            this.do_warn(_t("Please provide valid email addresses"), "");
+        }
+    },
+    on_user_clicked: function (event) {
+        var self = this;
+        var user_id = $(event.target).data('user-id');
```
Comment (mge-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: @pga-odoo  Why target and not currentTarget ?

File: addons/web_dashboard/static/src/js/dashboard.js [hunk 17]
```diff
@@ -0,0 +1,264 @@
...
+            return re.test(email);
+        });
+        var optional_message = this.$('#optional_message').val();
+        if(is_valid_emails){
+            $target.prop('disabled', true);
+            $target.find('i.fa-cog').removeClass('hidden');
+            new Model("res.users")
+                .call("web_dashboard_create_users", [user_emails, optional_message])
+                .then(function (result) {
+                    self.reload();
```
Comment (mge-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: @pga-odoo result not important ?

File: addons/web_dashboard/static/src/js/dashboard.js [hunk 18]
```diff
@@ -0,0 +1,264 @@
...
+    },
+    share_linkedin: function(){
+        var popup_url = _.str.sprintf('http://www.linkedin.com/shareArticle?mini=true&url=%s&title=I am using odoo&summary=%s&source=www.odoo.com', encodeURIComponent(this.share_url), this.share_text);
+        this.sharer(popup_url);
+    },
+    sharer: function(popup_url){
+        window.open(
+            popup_url,
+            'Share Dialog',
+            'width=626,height=436');
```
Comment (mge-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: @pga-odoo Are you sure about this fixed size ? Not sure it's mobile friendly.

File: addons/web_dashboard/static/src/xml/dashboard.xml [hunk 19]
```diff
@@ -0,0 +1,169 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<templates xml:space="preserve">
```
Comment (mge-odoo) [inline addons/web_dashboard/static/src/xml/dashboard.xml]: @pga-odoo Wy do you need to preserve space ?

File: addons/web_dashboard/static/src/xml/dashboard.xml [hunk 20]
```diff
@@ -0,0 +1,169 @@
...
+            </div>
+        </div>
+    </t>
+
+    <t t-name="DashboardApps">
+        <i class="fa fa-cog fa-4x text-muted"></i>
+        <div class="o_web_dashboard_header">
+            <t t-if="installed_apps">
+                <t t-esc="installed_apps"></t>
+                <t t-if="installed_apps == 1">Installed App</t>
```
Comment (mge-odoo) [inline addons/web_dashboard/static/src/xml/dashboard.xml]: Usually we do not care about this detail.

File: addons/web_dashboard/static/src/xml/dashboard.xml [hunk 21]
```diff
@@ -0,0 +1,169 @@
...
+            <div class="o_invitation_form">
+                <strong>Invite new users</strong> <small class="text-muted">(with default rights)</small>
+                <textarea rows="3" id="user_emails" placeholder="Email addresses (one per line)"></textarea>
+                <div class="optional_message_toggler"><a style="vertical-align: top;"> Add optional message</a></div>
+                <textarea class="optional_message" id="optional_message" rows="6" placeholder="Optional message" style="display: none;"></textarea>
+                <button class="btn btn-primary btn-block o_send_invitations" role="button"> <strong><i class="fa fa-cog fa-spin hidden"/> Send Invitations</strong></button>
+            </div>
+            <div>
+                <small class="text-muted text-center">
+                    <t t-if="users.pending.length != 0">
```
Comment (mge-odoo) [inline addons/web_dashboard/static/src/xml/dashboard.xml]: ! users.pending.length

File: addons/web_dashboard/static/src/xml/dashboard.xml [hunk 22]
```diff
@@ -0,0 +1,169 @@
...
+            <div>
+                <small class="text-muted text-center o_compact_subtitle">
+                    Get a <strong><a href="http://www.amazon.com/gift-cards/" target="_new">$50 Amazon Gift Card</a></strong> for each customer who subscribes to Odoo through your referral link.
+                </small>
+                <hr/>
+            </div>
+            <strong >Copy &amp; Share this url:</strong>
+            <div class="mt16 alert alert-warning">
+                <span class="o_dashboard_sharearea" title="Click here to copy share URL" t-att-data-clipboard-text="widget.share_url"><t t-esc="widget.share_url"/></span>
+                <span class="o_clipbord_success_alert" style="display:none" t-att-data-clipboard-text="widget.share_url"><i class="fa fa-link"></i> URL copied to clipboard.</span>
```
Comment (mge-odoo) [inline addons/web_dashboard/static/src/xml/dashboard.xml]: @pga-odoo We try to avoid inline style. Why don't you put it in your less ?

File: addons/web_dashboard/static/src/js/dashboard.js (hunk 15 above)
Comment (pga-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: App Dashboard(first one) is just simple informative dashboard and it don't having much thing to do unlike other dashboards.

File: addons/web_dashboard/static/src/js/dashboard.js (hunk 16 above)
Comment (pga-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: Element on which event is attached don't having any child so it doesn't matter `target or currentTarget` because both going to return same result

File: addons/web_dashboard/static/src/js/dashboard.js (hunk 18 above)
Comment (pga-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: yes, because mobile browsers don't have pop up window or dialogs, they just open it in new tab

File: addons/web_dashboard/models/res_users.py (hunk 11 above)
Comment (mge-odoo) [inline addons/web_dashboard/models/res_users.py]: @pga-odoo What about this ? You decided to keep it in res.users ?

File: addons/web_dashboard/static/src/js/dashboard.js (hunk 18 above)
Comment (mge-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: Ok. And why these sizes ?

File: addons/web_dashboard/static/src/xml/dashboard.xml [hunk 23]
```diff
@@ -0,0 +1,169 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<templates xml:space="preserve">
+    <t t-name="DashboardMain">
+        <div class="o_web_dashboard">
```
Comment (mge-odoo) [inline addons/web_dashboard/static/src/xml/dashboard.xml]: container missing

File: addons/web_dashboard/static/src/js/dashboard.js (hunk 18 above)
Comment (pga-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: In chrome if i don't give height-width it open it in new tab, instead window. 
If i give height-width 1-1 it is like twitter window in screenshot. 
I just gave approx width and height (yes randomly !!) because facebook and linkedin changes window height according to content. I will change it with some proper number like 600*400  :smile:  
screenshot from 2015-08-27 18 15 45

File: addons/web_dashboard/static/src/js/dashboard.js (hunk 17 above)
Comment (pga-odoo) [inline addons/web_dashboard/static/src/js/dashboard.js]: No, because it always return same result and we just need to reload that dashboard when all operations are performed

File: addons/web_dashboard/static/src/xml/dashboard.xml (hunk 23 above)
Comment (pga-odoo) [inline addons/web_dashboard/static/src/xml/dashboard.xml]: Don't need container here. Fixed width change design of dashboard.

File: addons/web_dashboard/models/res_users.py (hunk 11 above)
Comment (pga-odoo) [inline addons/web_dashboard/models/res_users.py]: @mge-odoo  still not sure. should i move both function back to controller ??

File: addons/web_dashboard/static/src/xml/dashboard.xml (hunk 23 above)
Comment (mge-odoo) [inline addons/web_dashboard/static/src/xml/dashboard.xml]: A row must always have one of its parent which is a container. It's not the case here, so there is something wrong.

File: addons/web_dashboard/models/res_users.py (hunk 11 above)
Comment (mge-odoo) [inline addons/web_dashboard/models/res_users.py]: @pga-odoo  Yes I will move it. But don't bother fixing the small issues I noticed, I'll fix them myself to be able to merge soon. If you can just answer my questions, it's ok :-)

File: addons/web_dashboard/static/src/xml/dashboard.xml (hunk 23 above)
Comment (pga-odoo) [inline addons/web_dashboard/static/src/xml/dashboard.xml]: It is creating this kind of behavior.
On higher resolution it leave lot's of extra space 
1st one is with container
download 9

File: addons/web_dashboard/models/res_users.py (hunk 11 above)
Comment (pga-odoo) [inline addons/web_dashboard/models/res_users.py]: ok :+1:  tell me any time if you need me to fix something

File: addons/web_dashboard/static/src/xml/dashboard.xml (hunk 23 above)
Comment (mge-odoo) [inline addons/web_dashboard/static/src/xml/dashboard.xml]: Maybe with container-fluid then ? It's just that the bootstrap grid requires a container as a parent of row. Because of that, you have to use some hacks but it's not the cleanest way.

Comment (pga-odoo) [inline addons/web_dashboard/static/src/xml/dashboard.xml]: Yes, container-fluid works fine

File: addons/web_dashboard/static/src/less/dashboard.less [hunk 24]
```diff
@@ -0,0 +1,122 @@
...
+                    margin-bottom: 8px;
+                }
+        }
+        .o_web_dashboard_pills {
+            margin-top: 10px;
+            i.fa {
+                vertical-align: middle;
+                font-size: 1.5em;
+            }
+            a {
```
Comment (sle-odoo) [inline addons/web_dashboard/static/src/less/dashboard.less]: don't use css selectors with html tags if you do not help the browser to tie break quickly by adding a class or using the ">" selector. you have this probleme multiple times in your less file here

File: addons/web_dashboard/models/res_users.py [hunk 25]
```diff
@@ -0,0 +1,43 @@
...
+        existing_users = self.search(['&', ('login', 'in', emails), '|', ('active', '=', True), ('active', '=', False)])
+
+        # Processing existing users
+        for user in existing_users:
+            if not user.login_date and not user.signup_valid:
+                user.action_reset_password()
+            else:
+                # Already a user, reactivated in case it is deactivated
+                user.active = True
+        new_emails = set(emails) - set(existing_users.mapped('login'))
```
Comment (mge-odoo) [inline addons/web_dashboard/models/res_users.py]: @pga-odoo  .mapped('login') ? Login is not an email address ...

File: addons/web_dashboard/models/res_users.py [hunk 26]
```diff
@@ -0,0 +1,43 @@
...
+        template_user_id = self.env['ir.config_parameter'].get_param('auth_signup.template_user_id', 'False')
+
+        # Processing new email addresses : create new users (Used template user)
+        for email in new_emails:
+            default_values = {'login': email, 'name': email, 'email': email, 'active': True}
```
Comment (mge-odoo) [inline addons/web_dashboard/models/res_users.py]: @pga-odoo  Can't we find better default values for name and login ?

File: addons/web_dashboard/models/res_users.py [hunk 27]
```diff
@@ -0,0 +1,43 @@
...
+            if template_user_id:
+                user = self.with_context(custom_message=optional_message, create_user=1).browse(int(template_user_id)).copy(default=default_values)
+            else:
+                user = self.with_context(custom_message=optional_message, create_user=1).create(default_values)
+            # If user haven't configured Outgoing mail server properly
+            if not user.signup_token:
```
Comment (mge-odoo) [inline addons/web_dashboard/models/res_users.py]: @pga-odoo I don't understand the link between signup_token & outgoing mail server ... Can you explain me please ?

File: addons/web_dashboard/models/res_users.py (hunk 26 above)
Comment (pga-odoo) [inline addons/web_dashboard/models/res_users.py]: Here we invite user just by email, we don't having anything else. 
download 6

File: addons/web_dashboard/models/res_users.py (hunk 27 above)
Comment (pga-odoo) [inline addons/web_dashboard/models/res_users.py]: I have instruction to double check whether invitation is really sent or not, so in odoo if mail is not sent due to wrong configure it cancel the sign up and set token false.
https://github.com/odoo/odoo/blob/master/addons/auth_signup/res_users.py#L99 
so here i used redirect warning to for configuration of mail server.(because users of  this dashboard will have access to it)
we can change it to normal warning like "Invitation not sent ! something went wrong contact administrator "

File: addons/web_dashboard/models/res_users.py (hunk 26 above)
Comment (pga-odoo) [inline addons/web_dashboard/models/res_users.py]: You can take look at this, because of user have email id in name i provided a way at sign up, so user can change his/her name at sign up(just one time)
https://github.com/odoo/odoo/pull/7288/files#diff-fe544dce57f31c7962edb0ce697b5431R198

File: addons/web_dashboard/models/res_users.py (hunk 25 above)
Comment (pga-odoo) [inline addons/web_dashboard/models/res_users.py]: we always displayed login as a email address. And user created through dashboard will have email in login.
download
But it is not in every case(e.g. user's demo data has admin, demo etc :grin: ). so we can change to email, but there will be chances of someone make change in email id because email is related field(partner_id.email)

Comment (mge-odoo) [inline addons/web_dashboard/models/res_users.py]: Ok, thanks for explanations. But even with this particular case, I prefer to use email.

File: addons/web_dashboard/models/res_users.py (hunk 26 above)
Comment (mge-odoo) [inline addons/web_dashboard/models/res_users.py]: Ok, sure.

File: addons/web_dashboard/models/res_users.py (hunk 27 above)
Comment (mge-odoo) [inline addons/web_dashboard/models/res_users.py]: Ok, thanks. The redirect message with configuration of mail server is a good thing, I just wanted to be sure that it cannot be related to another problem.

**Comment by hmo-odoo** [comment]
merged at https://github.com/odoo/odoo/commit/c6a0af45efb6b30fc67af6645a2d71d58e8f4065
